app/models/manageiq/providers/hawkular/inventory/parser/middleware_manager.rb

```diff
@@ -1,94 +1,77 @@
 module ManageIQ::Providers
   module Hawkular
-    class MiddlewareManager::RefreshParser
+    class Inventory::Parser::MiddlewareManager < ManagerRefresh::Inventory::Parser
       include ::Hawkular::ClientUtils
 
-      def self.ems_inv_to_hashes(ems, options = nil)
-        new(ems, options).ems_inv_to_hashes
-      end
-
-      def initialize(ems, _options = nil)
-        @ems = ems
-        @eaps = []
-        @data = {}
+      def initialize
         @data_index = {}
       end
 
-      def ems_inv_to_hashes
-        # the order of the method calls is important here, because they make use of @eaps and @data_index
+      def parse
+        # the order of the method calls is important here, because they make use of @data_index
```

Comment:
I was able to remove all `@data_index` usages but one! The remaining one is tricky because it indexes by `name` rather than by `manage_uuid`/`ems_ref`. In order to remove this remaining usage, I would either a need `find_by` method in `InventoryObject` or to find a way to use `ems_ref` instead of `name`. The former may require some work in order to be efficient. The later, all depends on the data returned by the provider and I wasn't able to think way to get or build the needed `ems_ref` to replace the indexing by `name`

There is on-going work on Hawkular side regarding how inventory is made. So, for now, I will keep this and try to remove it in another PR once Hawkular changes are ready. Unfortunately, `name` is unique only in certain conditions and keeping this single usage of `@data_index` also enforces some order when fetching inventory :disappointed: